### PR TITLE
chore: wrapper overloads signatures

### PR DESCRIPTION
**Pull request by cexbrayat**
This provides a better type checking than the optional parameter.

ping @JessicaSachs as we talked about that on Discord: if you want to take a quick look, my explanation will probably make more sense with the proper code :)

### CODE REVIEW

File: src/dom-wrapper.ts [hunk 1]
```diff
@@ -10,19 +10,23 @@ export class DOMWrapper<ElementType extends Element> {
...
     return Array.from(classes)
   }
 
-  attributes(key?: string) {
+  attributes(): { [key: string]: string }
+  attributes(key: string): string
+  attributes(key?: string): { [key: string]: string } | string {
     const attributes = this.element.attributes
-    const attributeMap = {}
+    const attributeMap: { [key: string]: string } = {}
```
Comment (lmiller1990) [inline src/dom-wrapper.ts:27]: I tend to do `Record<string, string>`; is there any benefit to one or the other?

File: src/dom-wrapper.ts [hunk 2]
```diff
@@ -10,19 +10,23 @@ export class DOMWrapper<ElementType extends Element> {
...
     for (let i = 0; i < attributes.length; i++) {
-      const att = attributes.item(i)
+      const att = attributes.item(i)!
```
Comment (lmiller1990) [inline src/dom-wrapper.ts:29]: If the attribute is missing, won't `att.localName` give "could not find `localName` of undefined?

File: src/dom-wrapper.ts (hunk 1 above)
Comment (pikax) [reply src/dom-wrapper.ts:27]: `Record<string, string>` implementation is pretty much that:
```ts
type Record<K extends keyof any, T> = {
    [P in K]: T;
};
```

Using this syntax allows greater control on the value, in this case `Record<string, string>` is clearer in this case

File: src/dom-wrapper.ts (hunk 2 above)
Comment (pikax) [reply src/dom-wrapper.ts:29]: we are iterating the attribute list, so I don't think is possible this is `undefined` at runtime

File: src/dom-wrapper.ts (hunk 1 above)
Comment (lmiller1990) [reply src/dom-wrapper.ts:27]: I agree, can we make this change @cexbrayat ? thanks!

File: src/dom-wrapper.ts (hunk 2 above)
Comment (lmiller1990) [reply src/dom-wrapper.ts:29]: Right 🤦

Comment (pikax) [reply src/dom-wrapper.ts:29]: I had to double check when I was reviewing, probably a forof would be a bit more explicit, but that's a bit nit picking :D

File: src/dom-wrapper.ts (hunk 1 above)
Comment (cexbrayat) [reply src/dom-wrapper.ts]: Sure, done!

**Comment by cexbrayat** [comment]
@lmiller1990 Updated according to your comment, PTAL

File: src/dom-wrapper.ts (hunk 2 above)
Comment (cexbrayat) [reply src/dom-wrapper.ts:29]: Yeah I agree. I had to double check too when I added this `!`.

Comment (cexbrayat) [reply src/dom-wrapper.ts]: It looks like we all agree, so I pushed an extra commit with a refactoring to use forof instead.

**Comment by lmiller1990** [comment]
Thanks!

**Pull request merged**